### PR TITLE
[CPDLP-1659] Add adjustment flow

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-1659

### Changes proposed in this pull request

* ECF statement page add new additional adjustment flow

### Guidance to review

* Login as a finance user
* Go to ECF financial statement that is not payable or paid
* Scroll down to additional adjustments section

### CODE REVIEW

File: app/components/finance/additional_adjustments/table.rb [hunk 1]
```diff
@@ -16,6 +16,10 @@ def initialize(statement:)
       def total_amount
         adjustments.sum(&:amount)
       end
+
+      def show_links?
```
[the remark this answers is not in the document]
Comment (cwrw) [reply app/components/finance/additional_adjustments/table.rb:20]: discussed in person: maybe this should only be for `output_fee` with `paid` only?

**Review by cwrw — COMMENTED** [review]
looking great, discussed in person a few things:
- attempting to change a value is still using the old values when going forward
- going back to main pages isn't clearing sessions when revisiting flows
thanks Mook!

File: app/models/finance/statement.rb [hunk 2]
```diff
@@ -73,4 +73,12 @@ def previous_statements
       .where(cpd_lead_provider:)
       .where("payment_date < ?", payment_date)
   end
+
+  def ecf?
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply app/models/finance/statement.rb:77]: Added 👍

File: app/controllers/finance/adjustments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        add_another_form: [:add_another],
+      )[:add_another_form] || {}
+    end
+
+    def set_statement
+      @statement = Finance::Statement.find(params[:statement_id])
+    end
+
+    def redirect_if_statement_not_open
+      return if @statement.output_fee && @statement.open?
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply app/controllers/finance/adjustments_controller.rb:61]: Added `Finance::Statement.adjustment_editable?`

File: app/components/finance/additional_adjustments/table.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mooktakim) [reply app/components/finance/additional_adjustments/table.rb:20]: Now delegating to `statement.adjustment_editable?`

**Review by cwrw — APPROVED** [review]
thanks Mook 🙌

**Pull request merged**